proselint/checks/__init__.py

```diff
@@ -1 +1,394 @@
+                )
+                for m in matches[idx_minority]
+            ]
+
+    return results
+
+
+def preferred_forms_check(  # noqa: PLR0913, PLR0917
+    text: str,
+    items: list,
+    err: str,
+    msg: str,
+    ignore_case: bool = True,
+    offset: int = 0,
+) -> list[ResultCheck]:
+    """Build a checker that suggests the preferred form."""
+    flags = re.IGNORECASE if ignore_case else 0
+    regex = r"[\W^]{}[\W$]"  # TODO: replace with enum below?
+
+    return [
+        (
+            m.start() + 1 + offset,
+            m.end() + offset,
+            err,
+            msg.format(item[0], m.group(0).strip()),
+            item[0],
+        )
+        for item in items
+        for r in item[1]
+        for m in re.finditer(regex.format(r), text, flags=flags)
+    ]
+    # TODO: can we speed up str.format() ?
+    #       fast-string? or do padding already in checks
+
+
+# PADDINGS, add more if needed
+class Pd(str, Enum):
+    disabled = r"{}"
+    # choose for checks with custom regex
+    whitespace = r"\s{}\s"
+    # -> req whitespace around (no punctuation!)
+    sep_in_txt = r"(?:^|\W){}[\W$]"
+    # req non-text character around
+    # -> finds item as long it is surrounded by any non-word character:
+    #       - whitespace
+    #       - punctuation
+    #       - newline ...
+
+
+def existence_check(  # noqa: PLR0913, PLR0917
+    text: str,
+    re_items: list,
+    err: str,
+    msg: str,
+    ignore_case: bool = True,
+    string: bool = False,  # todo: why not default on?
+    offset: int = 0,  # todo: some checks set this strangely
+    padding: Pd = Pd.sep_in_txt,
+    dotall: bool = False,
+    excluded_topics: Optional[list] = None,
+    exceptions=(),
+) -> list[ResultCheck]:
+    """Build a checker that prohibits certain words or phrases."""
+    flags = 0
+    if ignore_case:
+        flags |= re.IGNORECASE
+    if string:
+        flags |= re.UNICODE
+    if dotall:
+        flags |= re.DOTALL
+
+    errors: list[ResultCheck] = []
+
+    # If the topic of the text is in the excluded list, return immediately.
+    if excluded_topics:
+        tps = topics(text)
+        if any(t in excluded_topics for t in tps):
+            return errors
+
+    rx = "|".join(padding.format(_item) for _item in re_items)
+    for m in re.finditer(rx, text, flags=flags):
+        txt = m.group(0).strip()
+        if any(re.search(exception, txt) for exception in exceptions):
+            continue
+        errors.append(
+            (m.start() + 1 + offset, m.end() + offset, err, msg.format(txt), None),
+        )
+        # TODO: doesn't the padding alter the start+end?
+
+    return errors
+
+
+def simple_existence_check(
+    text: str, pattern: str, err: str, msg: str, offset: int = 0
+):
+    """Build a checker for single patters.
+    in comparison to existence_check:
+        - does not work on lists
+        - no padding
+        - excluded topics or exceptions
+
+        TODO: maybe add ignorecase
+    """
```

Comment:
```suggestion
    """Builds a checker for simple patterns.
    Compared with existence_check, this:
        - does not work on lists
        - has no padding
        - excludes topics and exceptions
    """
    # TODO: add ignorecase?
```
While I'm here, can we change `build` to `builds` in the other docstrings?

Reply:
done